src/libPMacc/examples/gameOfLife2D/include/GatherSlice.hpp

```diff
@@ -20,18 +21,15 @@
  */
 
 
-#ifndef GATHERSLICE_HPP
-#define	GATHERSLICE_HPP
+#pragma once
 
-#include "types.h"
 
 
 #include <mpi.h>
 #include "mappings/simulation/GridController.hpp"
-
-//c includes
 #include "memory/boxes/PitchedBox.hpp"
 #include "dimensions/DataSpace.hpp"
+#include "types.h"                                  // DIM*
```

Comment:
Maybe local to global does not describe it perfectly.

Imagine the files and their includes as graph with the files as nodes and the includes they contain as directed edges to the other file nodes. The more incoming edges a file has, the more general it is (global in the inperfect description). 
The more incoming edges, a file has, the lower its position in the list of includes.
This prevents (to some degree) the propability that an inclusion of e.g. a standard header is missing in a project internal file because it is always included by another project internal file that always includes the standard header first.

`types.h` therefore is more general (global) than the other two.